Optimize string manipulation while logging

PaulZhang12 opened the pull request:
Summary: Inductor registers a logging handler, and converting absolute paths to relative paths turns out to be surprisingly expensive. Adding a cache makes `make_module_path_relative` about 120x faster and speeds up top level `inductor_compile` by 6-9%.

Test Plan:
```
tlp buck run mode/opt //scripts/paulzhan:repro
```
and then looking at perfetto.

Differential Revision: D88401486

Skylion007 reviewed (approved):
Please add a return type to the other function to so it forwards the str return

robieta committed:
Optimize string manipulation while logging (#169898)

Summary:

Inductor registers a logging handler, and converting absolute paths to relative paths turns out to be surprisingly expensive. Adding a cache makes `make_module_path_relative` about 120x faster and speeds up top level `inductor_compile` by 6-9%.

Test Plan:
```
tlp buck run mode/opt //scripts/paulzhan:repro
```
and then looking at perfetto.

Reviewed By: PaulZhang12

Differential Revision: D88401486

robieta commented:
@pytorchbot merge
